### ISSUE TITLE
Question for the inference results on PanNuke dataset

### ISSUE THREAD

**Issue by AlfredQin** [computational-cell-analytics/patho-sam#40]
Hi, @anwai98 , thanks for your promising work. I tried to test the provided weights of uSAM and PathoSAM on the fold 3 of PanNuke dataset using the codebase in Segment Anything for Microscopy. But I find that the MSA metric of uSAM is close to the paper but the PathoSAM has worse performance than uSAM in terms of interactive segmentation. I am not sure if there is some wrong in my code, which is shown below:
```python
  import os
  import glob
  import numpy as np
  from PIL import Image
  from micro_sam.util import get_sam_model
  from micro_sam.evaluation import inference
  from micro_sam.evaluation.model_comparison import _enhance_image
  from micro_sam.sample_data import fetch_tracking_example_data, fetch_tracking_segmentation_data
  from micro_sam.evaluation.evaluation import run_evaluation, run_evaluation_for_iterative_prompting
  
  root_dir = os.getcwd()
  
  def filter_no_foreground_samples(image_paths, gt_paths):
      filtered_image_paths = []
      filtered_gt_paths = []
      for image_path, gt_path in zip(image_paths, gt_paths):
          gt_mask = Image.open(gt_path).convert('L')
          gt_mask = np.array(gt_mask)
          if np.sum(gt_mask) > 0:  # Check if there are any foreground pixels
              filtered_image_paths.append(image_path)
              filtered_gt_paths.append(gt_path)
      return filtered_image_paths, filtered_gt_paths
  
  checkpoint=None  # overwrite in case you have custom checkpoints to load models from
  model_type = "vit_b_histopathology"  # overwrite with your desired choice of model
  experiment_folder = os.path.join(root_dir, "experiment_dir", "PanNuke", model_type)
  
  predictor = get_sam_model(model_type=model_type, checkpoint_path=checkpoint)
  prediction_dir = os.path.join(experiment_folder, "start_with_point_prompt")
  embedding_folder = os.path.join(experiment_folder, "embeddings")
  
  test_image_dir = "the path of PanNuke dataset/Fold 3/images"
  test_gt_dir = "the path of PanNuke dataset/Fold 3/inst_masks"
  test_image_paths = sorted(glob.glob(os.path.join(test_image_dir, "*.png")))
  test_gt_paths = sorted(glob.glob(os.path.join(test_gt_dir, "*.png")))
  test_image_paths, test_gt_paths = filter_no_foreground_samples(test_image_paths, test_gt_paths)
  
  inference.run_inference_with_iterative_prompting(
      predictor=predictor,
      image_paths=test_image_paths,
      gt_paths=test_gt_paths,
      embedding_dir=embedding_folder,
      prediction_dir=prediction_dir,
      start_with_box_prompt=False,  # overwrite to start with the box prompt
      use_masks=False,  # overwrite to use logits masks for iterative prompting
  )
  run_evaluation_for_iterative_prompting(
      gt_paths=test_gt_paths,
      prediction_root=prediction_dir,
      experiment_folder=experiment_folder,
      start_with_box_prompt=False,  # overwrite th evaluate the iterative prompting starting with box
  )
  
  prediction_dir = os.path.join(experiment_folder, "start_with_box_prompt")
  embedding_folder = os.path.join(experiment_folder, "embeddings")
  
  inference.run_inference_with_iterative_prompting(
      predictor=predictor,
      image_paths=test_image_paths,
      gt_paths=test_gt_paths,
      embedding_dir=embedding_folder,
      prediction_dir=prediction_dir,
      start_with_box_prompt=True,  # overwrite to start with the box prompt
      use_masks=False,  # overwrite to use logits masks for iterative prompting
  )
  
  run_evaluation_for_iterative_prompting(
      gt_paths=test_gt_paths,
      prediction_root=prediction_dir,
      experiment_folder=experiment_folder,
      start_with_box_prompt=True,  # overwrite the evaluate the iterative prompting starting with box
  )
```
Could you please help double check it ? Besides, is the vit_b_histopathology  the released checkpoint of PathoSAM since I do not find the checkpoint of PathoSAM on the current repository.

**Comment by anwai98** [comment]
Hi @AlfredQin.

Thanks for your interest in `patho-sam` and `micro-sam`.

I will try to address your questions one-by-one.

> But I find that the MSA metric of uSAM is close to the paper but the PathoSAM has worse performance than uSAM in terms of interactive segmentation.

With your current code, your speculation is right (I'll explain below)

> I am not sure if there is some wrong in my code, which is shown below:

Overall, your code on a higher level looks okay. In any case, I created a script [here](https://github.com/computational-cell-analytics/patho-sam/blob/master/scripts/test_pannuke.py) just now (as I was curious and wanted to reproduce your observation)

> Besides, is the vit_b_histopathology the released checkpoint of PathoSAM since I do not find the checkpoint of PathoSAM on the current repository.

You are right. `patho-sam` has been integrated in `micro-sam` now. With this, the `patho-sam` models are directly compatible in the `micro-sam` workflow. And you can access `patho-sam` models under the `vit_b_histopathology` name! (and good catch actually, I'll mention this somewhere in our doc)

With the scripts I provided, if you look closely, I pad up the PanNuke images to shape (512, 512). With this, I can reproduce the expected results using `patho-sam` model. This is done to match the patch shape on which the model has been trained. Since the PanNuke images are small crops of shape (256, 256), and when this gets upsized to (1024, 1024) by Segment Anything, the objects have a changed appearance, leading to the tiny drop in interactive segmentation performance.

The suggested script should do the trick for you. Let me know how it works!

**Comment by anwai98** [comment]
I'll close this issue due to inactivity. Please feel free to create a new issue if you come across any issues!
